src/components/CGroupTable.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <div class="px-4 pt-2">
+    <div class="flex float-right w-48 mb-3 search relative mr-6 my-2">
+      <div class="absolute pin-r pin-t mt-3 ml-2 mr-4 text-purple-lighter">
+        <img src="@/assets/img/lupe.svg" width="17rem" height="20rem" />
+      </div>
+      <input
+        class="placeholder-gray-700 bg-gray-100 mr-2 text-sm font-medium leading-5  focus:outline-none  py-2 px-2 block w-full appearance-none leading-normal"
+        id="filter"
+        placeholder="Search..."
+        type="search"
+        style="text-indent:20px"
+        v-model.lazy="filter"
+      />
+      <div class="loader absolute right-0" v-if="isLoading">Loading...</div>
+    </div>
+    <table class="w-full bg-white rounded mb-4">
+      <thead>
+        <tr class="bg-gray-100 h-8">
+          <c-th
+            v-for="(column, index) in columns"
+            :label="column.label"
+            :name="column.name"
+            :key="index"
+          ></c-th>
+        </tr>
+      </thead>
+      <tbody>
+        <template v-for="g in groups">
+          <c-group-header
+            :key="g.groupId + 'header'"
+            :groupId="g.groupId"
+            :messages="g.messages"
+            @onToggle="toggleRow"
+          ></c-group-header>
+          <c-group-content
+            v-if="openedRows.includes(g.groupId)"
+            :key="g.groupId + 'content'"
+            :messages="g.messages"
+            :groupId="g.groupId"
+          ></c-group-content>
+        </template>
+        <tr class="border text-xs h-10 text-gray-800" v-if="countGroups > 10">
+          <td :colspan="columns.length">
+            <c-page-footer :total="countGroups"></c-page-footer>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+<script>
+import CGroupContent from '@/components/CGroupContent';
+import CGroupHeader from '@/components/CGroupHeader';
+import CPageFooter from '@/components/CPageFooter';
+import CTh from '@/components/CTh';
+import utils from '@/utils';
+import { mapState } from 'vuex';
+
+export default {
+  name: 'CGroupTable',
+  components: { CTh, CGroupHeader, CGroupContent, CPageFooter },
+  data() {
+    return {
+      columns: [
+        { label: 'Group Id', name: 'groupId' },
+        { label: 'Actors' },
+        { label: 'Message Count' },
+        { label: 'Enqueued Datetime' },
+        { label: 'Progress', name: 'progress' },
+        { label: 'Remaining Time', name: 'remainingTime' }
+      ],
+      openedRows: []
+    };
+  },
+  methods: {
+    toggleRow(id) {
+      this.openedRows = utils.toggleItemFromList(id, this.openedRows);
+    }
+  },
+  computed: {
+    ...mapState(['groups', 'refreshInterval', 'isLoading', 'countGroups']),
+    filter: {
```

Comment:
it is necessary to delegate filter behaviours and attributes, to a component filter, in order to avoid duplicated code